### ISSUE TITLE
checknetwork return ok if server with different namespace is found

### ISSUE THREAD

**Issue by barbalberto** [robotology/yarp#486]
Test case:
```bash
yarp namespace /foo
yarpserver
```

```bash
yarp namespace /bar
yarp detect  #-> fails: not found
yarpscope #(or other yarp executables using the following check)
```

```c++
    //Yarp network initialization
    yarp::os::Network yarp;
    if (!yarp.checkNetwork()) {
        fatal() << "Cannot connect to yarp network";
    }
```
returns true and goes on but prints
```
warning: YARP name server(s) not configured, ports will be anonymous
warning: check your namespace and settings with 'yarp detect'
```

In this configuration, when we run yarp detect the yarpserver prints out
```
  * NAME_SERVER query "/bar"
 ASKED NAME_SERVER query /bar, GAVE *** end of message
  * NAME_SERVER query "/bar"
 ASKED NAME_SERVER query /bar, GAVE *** end of message
  * NAME_SERVER query "/bar" 
 ASKED NAME_SERVER query /bar, GAVE *** end of message
```
while in the correct configuration it is quiet.

If no yarpserver is running the check returns false and the program exits.
Is this the expected behaviour or the yarp.checkNetwork() should fail?
What does "ports will be anonymous" mean in this case?

@paulfitz @lornat75 @drdanz

**Comment by apaikan** [comment]
@barbalberto  Ok. I repeated your tests and got the same behavior. Actually that is the correct behavior. Let me clarify some parts about the Yarp network: 

1- Yarp network does not necessarily needs a `yarpserver` to be functional. for example: 

```
$ yarp write /127.0.0.1:10001
$ yarp read /127.0.0.1:10002
$ yarp connect /127.0.0.1:10001 /127.0.0.1:10002
```

2- So if the yarp network is not configure for the current namespace (e.g., in your example is the `/bar`) then the  `yarp.checkNetwork()` returns `true` which is correct. This means that you configured the Yarp network to work without any name server.  This is because that your `~/.config/yarp/_bar.conf` is empty (i.e., does not exists). 

3- If you are in the namespace which has already configured (i.e. there is a `~/.config/yarp/_foo.conf` for that in which the IP and port of the yarpserver is specified) then, this means that you force the yarp network to use a name server at the desired IP. In this case `yarp.checkNetwork()` returns `false` if the server cannot be find at that IP address. 
  
In short, changing to a new namespace WITHOUT configuring it does not mean that you are asking Yarp to use a `yarpserver` with  the desired namespace.
